go/protocol/keybase1/extras.go

```diff
@@ -1876,6 +1876,49 @@ func PublicKeyV1FromDeviceKeyV2(keyV2 PublicKeyV2NaCl) PublicKey {
 	}
 }
 
+func (dt *DeviceTypeV2) UnmarshalJSON(deviceType []byte) error {
+	deviceTypeStr := Unquote(deviceType)
+	switch deviceTypeStr {
+	// these values need to match the ones that the enum is defined on. We
+	// decode both numeric and lowercase text values (and both "backup" and
+	// "paper" for paper keys) as valid for backwards compatibility and
```

Comment:
I'm curious what the rationale for renaming to "paper" is - it seems to cause at least a mild amount of increased complexity.

Reply:
My rationale is that I wanted the constant to be called "paper" and not "backup" as it is more clear what a "paper device" is than what a "backup device" is. Unfortunately since the avdl compiler autogenerates constants value that means we have to deal with the added complexity.
Even if we left it as backup, the enum are by default marshalled to json as integers rather than strings, so some of the complexity would remain.
I am happy to change it if you believe this is too error prone or whatever, but this solution does not look too bad to me.

Reply:
Keeping seems reasonable to me - thanks for the explanation